src/messages.cpp

```diff
@@ -778,16 +778,19 @@ namespace quicr::messages {
         Bytes payload;
 
         payload << UintVar(msg.subscribe_id);
-        payload << msg.track_namespace;
-        payload << UintVar(msg.track_name.size());
-        payload << msg.track_name;
         payload.push_back(msg.priority);
         auto group_order = static_cast<uint8_t>(msg.group_order);
         payload.push_back(group_order);
+        payload << UintVar(msg.fetch_type);
+        payload << msg.track_namespace;
+        payload << UintVar(msg.track_name.size());
+        payload << msg.track_name;
         payload << UintVar(msg.start_group);
         payload << UintVar(msg.start_object);
         payload << UintVar(msg.end_group);
         payload << UintVar(msg.end_object);
+        payload << UintVar(msg.joining_subscribe_id);
```

Comment:
`joining_subscribe_id` and `preceding_group_offset` are only present if `fetch_type` == `kJoiningFetch`.